compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1872,6 +1872,13 @@ rustc_queries! {
         eval_always
         desc { "fetching all foreign CrateNum instances" }
     }
+    // Crates that are loaded non-speculatively (not for diagnostics or doc links).
+    // FIXME: This is currently only used for collecting lang items, but should be used instead of
+    // `crates` in most other cases too.
```

Comment:
Can we describe what situations would require using `crates` instead or is there not a common theme?

Reply:
Metadata encoding at least, if the second item in https://github.com/rust-lang/rust/pull/121167#issuecomment-1947413815 is not fixed.
If metadata includes unused crates, then SVH and binary depinfo should probably include them either.

Any diagnostic logic, like `trimmed_def_paths` may include them as well, because why not.